### PR TITLE
[front] - enh(assistant-builder): vaults empty

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at handling more nicely the data selection step within the assistant builder. When a managed data source is empty we display a button to enable the addition/request of data directly within the modal.

**References:**
- https://github.com/dust-tt/tasks/issues/1340

## Risk

Low

## Deploy Plan

Deploy `front`

### CODE REVIEW

**Comment by JulesBelveze** [comment]
Note: most files are edited to be able to pass params around..

**Review by Fraggle — APPROVED** [review]
nit: can you avoid props drilling by passing plan & dustClientFacingUrl via the AssistantBuilderContext ?

File: front/components/DataSourceViewResourceSelectorTree.tsx [hunk 1]
```diff
@@ -80,6 +86,7 @@ function DataSourceViewResourceSelectorChildren({
   showExpand,
   viewType = "documents",
 }: DataSourceResourceSelectorChildrenProps) {
+  const { plan, dustClientFacingUrl } = useContext(AssistantBuilderContext);
```
Comment (Fraggle) [inline front/components/DataSourceViewResourceSelectorTree.tsx:89]: that being said, becareful, i think the same component is used in vaults datasourceview selection (without the context).
You might want to test.

Comment (JulesBelveze) [reply front/components/DataSourceViewResourceSelectorTree.tsx:89]: You right good catch, will fix. Thanks for the heads up 👍🏼

**Pull request merged**